builder/azure/arm/config.go

```diff
@@ -401,6 +401,13 @@ type Config struct {
 	// See [documentation](https://docs.microsoft.com/en-us/azure/virtual-machines/user-data)
 	// to learn more about user data.
 	UserData string `mapstructure:"user_data" required:"false"`
+
+	// Specify a command to inject into the CustomScriptExtension, to run on startup
```

Comment:
This only works for Windows VMs because Linux uses a different VM extension (see https://learn.microsoft.com/en-us/azure/virtual-machines/extensions/custom-script-linux), given that linux has SSH out of the box I don't think we necessarily need to support Linux right now for running script at launch, but I think we should add a validation case this file to throw a validation error if custom_script is set for a linux build, what do you think?

Reply:
Done 👍